sklearn/metrics/classification.py

```diff
@@ -179,6 +179,80 @@ def accuracy_score(y_true, y_pred, normalize=True, sample_weight=None):
     return _weighted_sum(score, sample_weight, normalize)
 
 
+def top_n_accuracy_score(y_true, y_pred, n=5, normalize=True):
+    """top N Accuracy classification score.
+    For multiclass classification tasks, this metric returns the
+    number of times that the correct class was among the top N classes
+    predicted.
+
+    Parameters
+    ----------
+    y_true : 1d array-like, or label indicator array / sparse matrix
+        Ground truth (correct) labels.
+
+    y_pred : array-like, where for each sample, each row represents the
+        likelihood of each possible label.
+        The number of columns must be at least as large as the set of possible
+        label values.
+
+    normalize : bool, optional (default=True)
+        If ``False``, return the number of top N correctly classified samples.
+        Otherwise, return the fraction of top N correctly classified samples.
+
+    Returns
+    -------
+    score : float
+        If ``normalize == True``, return the proportion of top N correctly
+        classified samples, (float), else it returns the number of top N
+        correctly classified samples (int.)
+
+        The best performance is 1 with ``normalize == True`` and the number
+        of samples with ``normalize == False``.
+
+    See also
+    --------
+    accuracy_score
+
+    Notes
+    -----
+    If n = 1, the result will be the same as the accuracy_score. If n is the
+    same as the number of classes, this score will be perfect and meaningless.
+
+    In cases where two or more classes are assigned equal likelihood, the
+    result may be incorrect if one of those classes falls at the threshold, as
+    one class must be chosen to be the nth class and the class chosen may not
+    be the correct one.
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import top_n_accuracy_score
+    >>> y_pred = np.array([[0.1, 0.3, 0.4, 0.2],
+    ...                     [0.4, 0.3, 0.2, 0.1],
+    ...                     [0.2, 0.3, 0.4, 0.1],
+    ...                     [0.8, 0.1, 0.025, 0.075]])
+    >>> y_true = np.array([2, 2, 2, 1])
+    >>> top_n_accuracy_score(y_true, y_pred, n=1)
+    0.5
+    >>> top_n_accuracy_score(y_true, y_pred, n=2)
+    0.75
+    >>> top_n_accuracy_score(y_true, y_pred, n=3)
+    1.0
+    >>> top_n_accuracy_score(y_true, y_pred, n=2, normalize=False)
+    3
+    """
```

Comment:
It might be useful to add some reference papers / textbooks in the docstring if possible.